### PR TITLE
[Console] [DX] Fix command description/help display

### PR DESCRIPTION
**Pull request by noniagriconomie**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no   
| Deprecations? | no 
| Tests pass?   | yes  
| Fixed tickets | https://github.com/symfony/symfony/issues/26376
| License       | MIT
| Doc PR        | none

Hi,

Here is a fix for the issue https://github.com/symfony/symfony/issues/26376

I only patch the text descriptor, should I do it for others?
(I do not think so as the main and default one is text)

Thanks

### CODE REVIEW

File: src/Symfony/Component/Console/Descriptor/TextDescriptor.php [hunk 1]
```diff
@@ -140,6 +140,13 @@ protected function describeCommand(Command $command, array $options = array())
         $command->getSynopsis(false);
         $command->mergeApplicationDefinition(false);
 
+        if ($description = $command->getDescription()) {
+            $this->writeText('<comment>Description:</comment>', $options);
+            $this->writeText("\n");
+            $this->writeText('  '.$description);
+            $this->writeText("\n\n");
+        }
+
```
Comment (noniagriconomie) [inline src/Symfony/Component/Console/Descriptor/TextDescriptor.php]: should as well the ```$command->getProcessedHelp()``` return only the ```$command->getHelp()``` and not sometimes the ```$command->getDescription()```?

Comment (chalasr) [reply src/Symfony/Component/Console/Descriptor/TextDescriptor.php]: I think it should yes, it's weird that `getProcessedHelp()` can return the description., but I'm not sure it's fine to change it now. Instead, could we display the `Help` section (calling `getProcessedHelp()`) only if `getHelp()` is not null? or only if `getProcessedHelp() !== getDescription()`?

Comment (noniagriconomie) [reply src/Symfony/Component/Console/Descriptor/TextDescriptor.php]: @chalasr ok, I will change the called method here

**Comment by chalasr** [comment]
@noniagriconomie For other descriptors, I also think they don't need to be updated. In fact, the TextDescriptor seems to be the only one that does not contain the description.

By the way, I must say that you're working on a part which is quite hard to update and not very "first-time-contribution friendly" (not contribution friendly at all in fact), and you just do it perfectly for now!
Congratz, welcome and thank you for contributing to Symfony :)

**Comment by noniagriconomie** [comment]
@chalasr PR code updated accordingly
Ok for others command descriptor :)

**Review by chalasr — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Console/Descriptor/TextDescriptor.php [hunk 2]
```diff
@@ -154,7 +161,8 @@ protected function describeCommand(Command $command, array $options = array())
             $this->writeText("\n");
         }
 
-        if ($help = $command->getProcessedHelp()) {
+        $help = $command->getProcessedHelp();
+        if ($help !== $command->getDescription()) {
```
[the remark this answers is not in the document]
Comment (noniagriconomie) [reply src/Symfony/Component/Console/Descriptor/TextDescriptor.php]: 👍 change is done

**Review by chalasr — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]
@noniagriconomie I love DX issues, so thanks a lot for contributing this one. And congrats on your first Symfony contribution!

**Comment by noniagriconomie** [comment]
@javiereguiluz hope this help a lot of sf devs :)
regards

**Comment by fabpot** [comment]
@noniagriconomie Thanks for the pull request. I don't think this qualifies as a bug. So, I'm going to merge it into master into 2.7.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @noniagriconomie.

**Comment by noniagriconomie** [comment]
@fabpot i agree with you, it is not a « bug », more like a mini feature
Moreover, i really think at least it could (should) be merged on the 3.4 lts minimum (for symfony dev experience)
It is weird that the description is displayed at the end
Anyway, thanks for your review/time

**Pull request closed without merging**